consensus/src/block_storage/sync_manager.rs

```diff
@@ -132,6 +174,44 @@ impl BlockStore {
         Ok(())
     }
 
+    // Before calling this function, we need to maintain an invariant that ordered_cert.commit_info().id()
+    // is already in the block store. So, currently insert_ordered_cert calls are preceded by insert_quorum_cert calls
+    // to ensure this.
+    pub async fn insert_ordered_cert(
+        &self,
+        ordered_cert: &WrappedLedgerInfo,
+        retriever: &mut BlockRetriever,
+    ) -> anyhow::Result<()> {
+        if self.ordered_root().round() < ordered_cert.ledger_info().ledger_info().round() {
+            if let Some(ordered_block) = self.get_block(ordered_cert.commit_info().id()) {
+                if !ordered_block.block().is_nil_block() {
+                    observe_block(
+                        ordered_block.block().timestamp_usecs(),
+                        BlockStage::OC_ADDED,
+                    );
+                }
+                SUCCESSFUL_EXECUTED_WITH_ORDER_VOTE_QC.inc();
```

Comment:
nit: I'll just call it order_cert instead of order_vote_qc

Reply:
Order cert could actually be a 2-chain QC or order vote QC. I explicitly want to say order vote QC.